### ISSUE TITLE
nan error: linear regression demo at tensorflow.org

### ISSUE THREAD

**Issue by ziaoang** [tensorflow/tensorflow#3148]
There is a linear regression demo on [Here](https://www.tensorflow.org/versions/r0.9/get_started/index.html).
The x_data is generate by
```python
x_data = np.random.rand(100).astype(np.float32)
```
but when I change it to
```python
x_data = np.arange(100).astype(np.float32)
```
the "w" and "b" will become "nan".

**Comment by aselle** [comment]
This type of question is best asked on StackOverflow. I don't believe this is a bug. Your arange() version has a mean of 50ish. That means the learning rate provided to gradient descent will be too high for stable convergence. You can fix this by 
x_data = np.arange(100).astype(np.float32) / 100
or if you don't want to change your data you'll need to reduce learning rate to .0001 and take 40000ish iterations.
